test/aleph/http_test.clj

```diff
@@ -1438,3 +1442,18 @@
                    :http-versions [:http1]})]
       (is (instance? IllegalArgumentException result))
       (is (= "use-h2c? may only be true when HTTP/2 is enabled." (ex-message result))))))
+
+(deftest test-in-flight-request-cancellation
+  (let [conn-established (promise)
+        conn-closed (promise)]
+    (with-tcp-server (fn [s _]
+                       (deliver conn-established true)
+                       ;; Required for the client close to be detected
+                       (s/consume identity s)
+                       (s/on-closed s (fn []
+                                        (deliver conn-closed true))))
+      (let [rsp (http-get "/")]
+        (is (= true (deref conn-established 1000 :timeout)))
+        (http/cancel-request! rsp)
+        (is (= true (deref conn-closed 1000 :timeout)))
+        (is (thrown? RequestCancellationException (deref rsp 1000 :timeout)))))))
```

Comment:
I'd like to see more testing, for both normal and error conditions. Maybe via overriding the conn-pool in some cases.

What's all the TCP handler/server/etc stuff get us? Why not the the HTTP code?

Reply:
> I'd like to see more testing, for both normal and error conditions. 

You mean for the case where `cancel-request!` is invoked after the request had already completed successfully or due to some other error?

> What's all the TCP handler/server/etc stuff get us? Why not the the HTTP code?

IMHO using a raw TCP server which only  accepts and holds the connection is the cleanest way to reliably test whether cancellation closes the underlying TCP connection (via the `s/on-closed` handler). One could probably concoct something like it with a `raw-stream?` HTTP request handler but I don't think it adds much value, does it?

Reply:
I'll give the latter idea a shot in any case!

Reply:
OK, the test uses the HTTP server(s) now, works just as well and probably causes less confusion :+1: